src/Std/Data/HashMap/RawLemmas.lean

```diff
@@ -705,6 +709,51 @@ theorem distinct_keys [EquivBEq α] [LawfulHashable α] (h : m.WF) :
     m.keys.Pairwise (fun a b => (a == b) = false) :=
   DHashMap.Raw.distinct_keys h.out
 
+@[simp]
+theorem length_toList [EquivBEq α] [LawfulHashable α] (h : m.WF) :
+    m.toList.length = m.size :=
+  DHashMap.Raw.Const.length_toList h.out
+
+@[simp]
+theorem isEmpty_toList [EquivBEq α] [LawfulHashable α] (h : m.WF) :
+    m.toList.isEmpty = m.isEmpty :=
+  DHashMap.Raw.Const.isEmpty_toList h.out
+
+theorem mem_toList_iff_getElem?_eq_some [LawfulBEq α] (h : m.WF)
+    {k : α} {v : β} :
+    (k, v) ∈ m.toList ↔ m[k]? = some v :=
+  DHashMap.Raw.Const.mem_toList_iff_get?_eq_some h.out
```

Comment:
Oh, it's here, it just didn't get copied to `Lemmas`.

Could this be a `simp` lemma? Actually, the next lemma might be even better, assuming in the presence of `LawfulBEq` that its RHS continues simplifying.

Reply:
I don't think that we can make `simp` turn `m.getKey? k = some k` into `True` easily in the lemma below because that is only true if `k` is in the map. So I would suggest marking both as `simp` and making sure that `simp` prefers the `LawfulBEq` version (I think this will be the case if we swap the order of the theorems).